service/src/test/java/com/ge/predix/acs/monitoring/CacheHealthIndicatorTest.java

```diff
@@ -28,75 +28,100 @@ public void testHealth(final CacheHealthIndicator cacheHealthIndicator, final St
+                        AcsMonitoringUtilities.HealthCode.ERROR },
+                { mockCacheWithExceptionWhileGettingInfo(new RuntimeException(), DecisionCacheHealthIndicator.class,
+                        MockCacheHealthCheckUtil.DECISION),
                         Status.DOWN, AcsMonitoringUtilities.HealthCode.ERROR },
-                { mockCacheWithExceptionWhileGettingInfo(new RuntimeException(), ResourceCacheHealthIndicator.class),
+                { mockCacheWithExceptionWhileGettingInfo(new RuntimeException(), ResourceCacheHealthIndicator.class,
+                        MockCacheHealthCheckUtil.RESOURCE),
                         Status.DOWN, AcsMonitoringUtilities.HealthCode.ERROR },
-                { mockCacheWithExceptionWhileGettingInfo(new RuntimeException(), SubjectCacheHealthIndicator.class),
+                { mockCacheWithExceptionWhileGettingInfo(new RuntimeException(), SubjectCacheHealthIndicator.class,
+                        MockCacheHealthCheckUtil.SUBJECT),
                         Status.DOWN, AcsMonitoringUtilities.HealthCode.ERROR } };
     }
 
     private static void setMockCacheInternalFields(final CacheHealthIndicator decisionCacheHealthIndicator,
-            final boolean cachingEnabled, final boolean healthCheckEnabled) {
-        ReflectionTestUtils.setField(decisionCacheHealthIndicator, "cachingEnabled", cachingEnabled);
+            final boolean cachingEnabled, final boolean healthCheckEnabled, final String flag) {
         ReflectionTestUtils.setField(decisionCacheHealthIndicator, "healthCheckEnabled", healthCheckEnabled);
+        if (flag.equals(MockCacheHealthCheckUtil.DECISION)) {
+            ReflectionTestUtils.setField(decisionCacheHealthIndicator, "decisionCachingEnabled", cachingEnabled);
+        } else if (flag.equals(MockCacheHealthCheckUtil.RESOURCE)) {
+            ReflectionTestUtils.setField(decisionCacheHealthIndicator, "resourceCachingEnabled", cachingEnabled);
+        } else if (flag.equals(MockCacheHealthCheckUtil.SUBJECT)) {
+            ReflectionTestUtils.setField(decisionCacheHealthIndicator, "subjectCachingEnabled", cachingEnabled);
+        }
     }
 
     @SuppressWarnings("unchecked")
     private static CacheHealthIndicator mockCache(final boolean cachingEnabled, final boolean healthCheckEnabled,
-            final Class clazz) throws Exception {
+            final Class clazz, final String flag) throws Exception {
         RedisConnectionFactory redisConnectionFactory = Mockito.mock(RedisConnectionFactory.class);
         CacheHealthIndicator decisionCacheHealthIndicator = (CacheHealthIndicator) clazz
                 .getConstructor(RedisConnectionFactory.class).newInstance(redisConnectionFactory);
-        setMockCacheInternalFields(decisionCacheHealthIndicator, cachingEnabled, healthCheckEnabled);
+        setMockCacheInternalFields(decisionCacheHealthIndicator, cachingEnabled, healthCheckEnabled, flag);
```

Comment:
It's not just decisionCacheHealthIndicator any more.  Should rename to cacheHealthIndicator